app/api/threads/get_thread.feature

```diff
@@ -47,57 +47,50 @@ Feature: Get thread
       | 50      | 3              | closed                  | 20              | 2020-01-10 00:00:00 |
     And the time now is "2020-01-20T00:00:00Z"
 
-  Scenario: Status should be not_started if the thread doesn't exists
+  Scenario: Should return all fields when the thread exists
     Given I am the user with id "1"
-    When I send a GET request to "/items/10/participant/1/thread"
+    When I send a GET request to "/items/21/participant/1/thread"
     Then the response code should be 200
-    And the response body should be, in JSON:
+    And the response body decoded as "ThreadTokenResponse" should be, in JSON:
     """
     {
       "participant_id": 1,
-      "item_id": 10,
-      "status": "not_started"
+      "item_id": 21,
+      "status": "waiting_for_trainer",
+      "token": {
+        "can_watch": false,
+        "can_write": true,
+        "date": "20-01-2020",
+        "exp": "1579485600",
+        "is_mine": true,
+        "item_id": "21",
+        "participant_id": "1",
+        "user_id": "1"
+      }
     }
     """
```

Comment:
As discussed previously, gherkin tests should be clear enough so that the reader knows how to interpret these without checking the rules. Here, I do not really know how to interpret that without reading the "ThreadTokenResponse" decoder (which is not very "readable").

Actually, as a reader, I would expect something like:
```
    And the response body should be, in JSON:
    """
    {
      "participant_id": 1,
      "item_id": 10,
      "status": "not_started"
      "item_id": 21,
      "status": "waiting_for_trainer",
      "token": $atoken
    }
    """
  And $atoken is a valid "ThreadToken" with in its payload:
   """
     {
        "can_watch": false,
        "can_write": true,
        "date": "20-01-2020",
        "exp": "1579485600",
        "is_mine": true,
        "item_id": "21",
        "participant_id": "1",
        "user_id": "1"
      }
   """
```
(or "$.token of the response is.... " and/or "is a valid JWT signed by the backend". 

Because the "token" is quite opaque for the frontend, which is mainly interested in knowing the attributes in the response, but not the token that it will just pass as a string.  The "AlgoreaForum" backend service which will receives the token is only interested by what it is supposed to find in the token. The format that you describe here is something completely fake as there is no "can_watch" key in the same "structure" as the `status` key.

Of course, that's just an opinion, let me know.

Reply:
I tried to use a wildcard <*> to replace the token but it makes the comparison with the actual response very difficult to do. I didn't even try: it involves replacing <*> with what we get in the actual response.

The easiest way is to define the token first, and replace it in the response. There's already an existing Gherkin feature for it.

It would be difficult to prefix the variable by $ because it would conflict with the reference system.

This is what I came up with:

```
  Scenario: Should return all fields when the thread exists
    Given I am the user with id "1"
    When I send a GET request to "/items/21/participant/1/thread"
    Then the response code should be 200
    And the following token "threadToken" signed by the app is distributed:
      """
        {
          "can_watch": false,
          "can_write": true,
          "date": "20-01-2020",
          "exp": "1579485600",
          "is_mine": true,
          "item_id": "21",
          "participant_id": "1",
          "user_id": "1"
        }
      """
    And the response body should be, in JSON:
      """
        {
          "participant_id": 1,
          "item_id": 21,
          "status": "waiting_for_trainer",
          "token": "{{threadToken}}"
        }
      """
```